src/kdtree.rs

```diff
@@ -137,6 +139,34 @@ impl<A: Float + Zero + One, T: std::cmp::PartialEq, U: AsRef<[A]> + std::cmp::Pa
             .collect())
     }
 
+    pub fn top_n_within_unsorted<F>(
```

Comment:
maybe we can refactor a bit to reduce duplication with `within` fn?
also would be nice to have test cases on only top n are returned